v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerShardedMigrationWithoutMigrationShardIdColumnIT.java

```diff
@@ -0,0 +1,286 @@
+      "DataStreamToSpannerShardedMigrationWithoutMigrationShardIdColumnIT/spanner-schema.sql";
+
+  private static HashSet<DataStreamToSpannerShardedMigrationWithoutMigrationShardIdColumnIT>
+      testInstances = new HashSet<>();
+  private static PipelineLauncher.LaunchInfo jobInfo1;
+  private static PipelineLauncher.LaunchInfo jobInfo2;
+
+  public static PubsubResourceManager pubsubResourceManager;
+  public static SpannerResourceManager spannerResourceManager;
+
+  /**
+   * Setup resource managers and Launch dataflow job once during the execution of this test class.
+   *
+   * @throws IOException
+   */
+  @Before
+  public void setUp() throws IOException {
+    // Prevent cleaning up of dataflow job after a test method is executed.
+    skipBaseCleanup = true;
+    synchronized (DataStreamToSpannerShardedMigrationWithoutMigrationShardIdColumnIT.class) {
+      testInstances.add(this);
+      if (spannerResourceManager == null) {
+        spannerResourceManager = setUpSpannerResourceManager();
+        createSpannerDDL(spannerResourceManager, SPANNER_DDL_RESOURCE);
+      }
+      if (pubsubResourceManager == null) {
+        pubsubResourceManager = setUpPubSubResourceManager();
+      }
+      if (jobInfo1 == null) {
+        jobInfo1 =
+            launchDataflowJob(
+                getClass().getSimpleName() + "shard1",
+                null,
+                null,
+                "shard1",
+                spannerResourceManager,
+                pubsubResourceManager,
+                new HashMap<>() {
+                  {
+                    put("inputFileFormat", "json");
+                  }
+                });
+      }
+      if (jobInfo2 == null) {
+        jobInfo2 =
+            launchDataflowJob(
+                getClass().getSimpleName() + "shard2",
+                null,
+                null,
+                "shard2",
+                spannerResourceManager,
+                pubsubResourceManager,
+                new HashMap<>() {
+                  {
+                    put("inputFileFormat", "json");
+                  }
+                });
+      }
+    }
+  }
+
+  /**
+   * Cleanup dataflow job and all the resources and resource managers.
+   *
+   * @throws IOException
+   */
+  @AfterClass
+  public static void cleanUp() throws IOException {
+    for (DataStreamToSpannerShardedMigrationWithoutMigrationShardIdColumnIT instance :
+        testInstances) {
+      instance.tearDownBase();
+    }
+    ResourceManagerUtils.cleanResources(spannerResourceManager, pubsubResourceManager);
+  }
+
+  @Test
+  public void multiShardMigration() {
+    // Two dataflow jobs are running corresponding to two physical shards containing two logical
+    // shards each. Migrates Users table from 4 logical shards. Asserts data from all the shards are
+    // going to Spanner. Checks whether migration shard id column is populated properly based on the
+    // transformation context.
```

Comment:
"Checks whether migration shard id column is populated properly based on the transformation context."
- Stale comment